specs/GetSecurityDescriptorForPackageFamilyNames/GetSecurityDescriptorForPackageFamilyNames.md

```diff
@@ -0,0 +1,49 @@
+# Easing and streamlining named object sharing between packaged and Win32
+
+This feature is a flat C API that takes a list of Package Family Names (PFNs) and access masks, and returns a security descriptor
+
+# Background
+
+Named objects (mutexes, events, waitable timers) must be ACLed during creation in order to be shared with packaged applications.
+
+Currently, [our guidance](https://docs.microsoft.com/en-us/windows/uwp/communication/sharing-named-objects) for this scenario is to write a large amount of complex, cumbersome, and largely boilerplate code. ([See here for a taste](https://docs.microsoft.com/en-us/windows/win32/api/securityappcontainer/nf-securityappcontainer-getappcontainernamedobjectpath#examples)).
+
+If we are to meet developers where they are, we need to reduce the friction of Win32 and UWP interop, and make it as easy as possible to share named objects.
+
+# Description
+
+Add a flat C API that takes an array of PFNs and access masks, and produces the needed `SECURITY_ATTRIBUTES` object.
```

Comment:
Can you also add a WinRT wrapper variant that returns the SDDL string form as well? Someone using C# can then use that with things like [System.Security.AccessControl](https://docs.microsoft.com/en-us/dotnet/api/system.security.accesscontrol.objectsecurity.setsecuritydescriptorsddlform?view=dotnet-plat-ext-3.1#System_Security_AccessControl_ObjectSecurity_SetSecurityDescriptorSddlForm_System_String_) or storing?

Reply:
Nice. I like that bridge to support C# through SDDL strings.

Reply:
Sounds like a good idea. I'll work on adding this to the spec, but I'm guessing this is going to be under a new top-level namespace. What do you think that should be?

Reply:
Microsoft.Security.AccessControl

Quite a few [Windows.Security.* namespaces already exist](https://docs.microsoft.com/uwp/api/?view=winrt-19041) but nothing quite on point as AccessControl

Reply:
I like it. Not quite as sure if I have time to get to that part right now, or if I need to push that out a couple weeks.

Reply:
Definitely don't have time to do it right now, since we still seem to be working out exactly how the different features will be organized. I've opened #201 to track the WinRT API, and will come back to this once we have a little more infrastructure in place.